### PR TITLE
Support custom CA

### PR DESCRIPTION
**Pull request by marians**
Fixes https://github.com/giantswarm/gsctl/issues/96

This change pulls in https://github.com/hashicorp/go-rootcerts to do several things:

- Allow for using a CA certificate contained in the Mac OS user keychain
- Allows for the use of the environment variable `GSCTL_CAFILE` to point to a custom CA bundle file
- Allows for the use of the environment variable `GSCTL_CAPATH` to point to a directory with custom CA certificates

## Testing

Requirements:

- An API endpoint using a certificate issued by a CA that is **not** available in your operating system keychain (Mac OS: Keychain "System Roots", other OSes: unknown).
- In addition, you need the CA certificate matching the used certificate in PEM format.

If you want to test, you can get both from me.

Build your test binary using the `make` approach: `make clean && make`. The resulting binary can be found in `./build/bin/gsctl-<os>-<arch>`.

### Environment variable GSCTL_CAFILE

1. Without env var

```
./build/bin/gsctl-<os>-<arch> ping --api-endpoint=<endpoint>
```

This should fail with error `x509: certificate signed by unknown authority`.

2. With env var:

```
GSCTL_CAFILE=/path/to/cacert.pem \
    ./build/bin/gsctl-<os>-<arch> \
    ping --api-endpoint=<endpoint>
```

This should succeed.

### Environment variable GSCTL_CAPATH

With env var:

```
GSCTL_CAPATH=/path/to/dir \
    ./build/bin/gsctl-<os>-<arch> \
    ping --api-endpoint=<endpoint>
```

where `/path/to/dir` is the path to a directory containing `cacert.pem` (and optionally other CA pem files.

### Keychain

1. Import the CA certificate to the Mac OS keychain or equivalent. (It is not required to configure the trust settings for the imported item.)

2. Run `./build/bin/gsctl-<os>-<arch> ping --api-endpoint=<endpoint>` again (without any of the above environment variables set.

This should succeed.

### CODE REVIEW

**Comment by marians** [comment]
Here are binaries for convenient testing:

[gsctl-0.6.1+git-windows-386.zip](https://github.com/giantswarm/gsctl/files/1091192/gsctl-0.6.1.git-windows-386.zip)
[gsctl-0.6.1+git-windows-amd64.zip](https://github.com/giantswarm/gsctl/files/1091193/gsctl-0.6.1.git-windows-amd64.zip)
[gsctl-0.6.1+git-linux-amd64.tar.gz](https://github.com/giantswarm/gsctl/files/1091195/gsctl-0.6.1.git-linux-amd64.tar.gz)
[gsctl-0.6.1+git-darwin-amd64.tar.gz](https://github.com/giantswarm/gsctl/files/1091194/gsctl-0.6.1.git-darwin-amd64.tar.gz)

File: commands/create_kubeconfig.go [hunk 1]
```diff
@@ -77,7 +78,10 @@ func createKubeconfig(cmd *cobra.Command, args []string) {
 		Timeout:   10 * time.Second,
 		UserAgent: config.UserAgent(),
 	}
-	apiClient := client.NewClient(clientConfig)
+	apiClient, clientErr := client.NewClient(clientConfig)
+	if clientErr != nil {
```
Comment (rossf7) [inline commands/create_kubeconfig.go]: Is this if needed?

Comment (marians) [inline commands/create_kubeconfig.go]: It was meant for error handling. But I forgot the ... error handling. ;-)

File: client/client.go [hunk 2]
```diff
@@ -39,9 +44,20 @@ func NewClient(clientConfig Configuration) *gsclientgen.DefaultApi {
 		configuration.Timeout = &clientConfig.Timeout
 	}
 
+	// set up client TLS so that custom CAs are accepted.
+	tlsConfig := &tls.Config{}
+	rootCertsErr := rootcerts.ConfigureTLS(tlsConfig, &rootcerts.Config{
```
Comment (rossf7) [inline client/client.go]: Good to see how this is done :) Not something I've had to do in go.

**Comment by oponder** [comment]
Just wanted to add a WFM on MacOS 👍

**Pull request merged**